utils/imagenet_topk_accuracy_driver.py

```diff
@@ -0,0 +1,332 @@
+def save_centered_cropped_dataset(validation_images_dir):
+    processed_validation_images_dir = os.path.join(validation_images_dir,
+                                                   "processed")
+    print "Saving centered cropped input images: %s" % \
+          processed_validation_images_dir
+
+    img_subdirs = get_sorted_img_subdirs(validation_images_dir)
+
+    try:
+        os.makedirs(processed_validation_images_dir)
+    except OSError:
+        sys.exit("New validation directory must not exist")
+
+    # Iterate over all labels subdirectories, loading, transforming and saving
+    # all images to the new location.
+    for img_subdir in img_subdirs:
+        orig_img_subdir_path = os.path.join(validation_images_dir, img_subdir)
+        processed_img_subdir_path = os.path.join(processed_validation_images_dir,
+                                                 img_subdir)
+
+        # Create a new subdirectory for the next label.
+        try:
+            os.makedirs(processed_img_subdir_path)
+        except OSError:
+            sys.exit("New label subdirectory somehow already existed.")
+
+        # Transform and save all images in this label subdirectory.
+        for orig_img_filename in os.listdir(orig_img_subdir_path):
+            orig_img_path = os.path.join(orig_img_subdir_path, orig_img_filename)
+            if os.path.isfile(orig_img_path):
+                processed_img_path = os.path.join(processed_img_subdir_path,
+                                                  orig_img_filename)
+                resize_and_save_image(orig_img_path, processed_img_path)
+
+## @returns a list of strings (of length equal to the @param batch_size) which
+# are paths to images to do inference on. @param img_paths is the set of all
+# image paths, @param img_index is the next index to use in @param img_paths,
+# and @param tmp_dir_name is the location of where to save the images if
+# @param resize_input_images is true.
+def get_curr_img_paths(img_paths, img_index, batch_size, tmp_dir_name,
+                       resize_input_images):
+    curr_img_paths = []
+    for batch_idx in xrange(batch_size):
+        img_path = img_paths[img_index + batch_idx]
+        # If we are resizing the image then we are going to save it to a
+        # temp location to read in later for inference.
+        if resize_input_images:
+            # Save the new image to the tmp directory.
+            path_to_tmp_img = os.path.join(tmp_dir_name, "tmp_img" +
+                                           str(batch_idx) + ".png")
+            resize_and_save_image(img_path, path_to_tmp_img)
+            img_path = path_to_tmp_img
+
+        curr_img_paths.append(img_path)
+
+    return curr_img_paths
+
+## Verifies that the @param image_classifier_cmd is well formatted via
+# assertions.
+def verify_spawn_cmd(image_classifier_cmd):
+    split_cmd = image_classifier_cmd.split()
+
+    assert "image-classifier" in split_cmd[0], \
```

Comment:
Using `\` for line breaks is discouraged by many style guides. You can use:
```
assert "image-classifier" in split_cmd[0], (
    "image-classifier binary must be provided first.")
```

Reply:
Done